lib/OpsGenieConnector.cpp

```diff
@@ -0,0 +1,28 @@
+#include "http_client.h"
+#include "OpsGenieConnector.h"
+
+const char *OpsGenieConnector::apiUrl = "https://api.opsgenie.com/v1/json/alert";
+
+OpsGenieConnector::OpsGenieConnector(const char *apiKey, const char *recipient) {
+    this->apiKey = apiKey;
+    this->recipient = recipient;
+}
+
+int OpsGenieConnector::pushAlert(const char *message) {
+    const char *json_format = "{\"message\":\"%s\", \"apiKey\": \"%s\", \"recipients\": [\"%s\"]}";
```

Comment:
I think it's confusing that you need to specify whom the alert is going to be routed to. I would expect this to be part of the `OpsGenie` configuration and me not worry about it as a `ProxySQL` user.